rockhound/bedmap2.py

```diff
@@ -0,0 +1,103 @@
+    :class:`xarray.Dataset` object.
+
+    Each dataset is projected in Antarctic Polar Stereographic projection, latitude of
+    true scale -71 degrees south, datum WGS84. All heights are in metres relative to
+    sea level as defined by the g104c geoid.
+
+    The available datasets are:
+
+    - ``bed``: bedrock height
+    - ``surface``: ice surface height
+    - ``thickness``: ice thickness
+    - ``icemask_grounded_and_shelves``: mask showing the grounding line and the extent
+      of the floating ice shelves
+    - ``rockmask``: mask showing rock outcrops
+    - ``lakemask_vostok``: mask showing the extent of the lake cavity of Lake Vostok
+    - ``grounded_bed_uncertainty``: ice bed uncertainty grid
+      (figure 12 of the manuscript)
+    - ``thickness_uncertainty_5km``: ice thickness uncertainty grid
+      (figure 11 of the manuscript)
+    - ``coverage``: binary grid showing the distribution of ice thickness data used
+      in the grid of ice thickness
+    - ``geoid``: values to convert from heights relative to WGS84 datum to heights
+      relative to EIGEN-GL04C geoid (to convert back to WGS84, add this grid)
+
+    Parameters
+    ----------
+    datasets : list
+        Datasets that will be loaded from the Bedmap2 model.
+    load : bool
+        Wether to load the data into an :class:`xarray.Dataset` or just return the
+        path to the downloaded data.
+
+    Returns
+    -------
+    grid : :class:`xarray.Dataset`
+        The loaded Bedmap2 datasets.
+    """
+    fname = REGISTRY.fetch("bedmap2_tiff.zip")
+    if not load:
+        return fname
+
+    for dataset in datasets:
+        if dataset not in DATASETS:
+            raise ValueError("Dataset {} not found in bedmap2_tiff.zip".format(dataset))
+    available_datasets = dict(
+        zip(DATASETS, ["bedmap2_{}.tif".format(dataset) for dataset in DATASETS])
+    )
+    available_datasets["geoid"] = "gl04c_geiod_to_WGS84.tif"
+    for i, dataset in enumerate(datasets):
+        with tempfile.TemporaryDirectory() as tempdir:
+            # Decompress the file into a temporary file so we can load it with xr
+            # The .tif files inside the zip are located inside a bedmap2_tiff directory
+            with ZipFile(fname, "r") as zip_file:
+                zip_file.extract(
+                    os.path.join("bedmap2_tiff", available_datasets[dataset]),
+                    path=tempdir,
+                )
+            array = xr.open_rasterio(
+                os.path.join(tempdir, "bedmap2_tiff", available_datasets[dataset])
+            )
+            # Replace no data values with nans
+            array = array.where(array != array.nodatavals)
+            # Remove "band" dimension and coordinate
+            array = array.squeeze("band", drop=True)
+            array.name = dataset
+            if i == 0:
+                grid = array.to_dataset()
+            else:
+                grid = xr.merge([grid, array])
```

Comment:
```python
    grids.append(array)
grid = xr.merge(grids)
```

Reply:
The metadata should be set after the merge because xarray doesn't carry the `attrs` over to the new grid.

Reply:
I thought that using lists would eat more memory, but I've just checked that it's the same, so I'll change the code according to your proposal (i.e. like I've implemented it on early stages).